shell/platform/darwin/ios/ios_external_texture_metal.mm

```diff
@@ -65,9 +68,123 @@
     return nullptr;
   }
 
+  sk_sp<SkImage> image = nullptr;
+  if (pixelFormat == kCVPixelFormatType_420YpCbCr8BiPlanarVideoRange ||
+      pixelFormat == kCVPixelFormatType_420YpCbCr8BiPlanarFullRange) {
+    image = WrapYUVAExternalPixelBuffer(pixel_buffer, context);
+  } else {
+    image = WrapRGBAExternalPixelBuffer(pixel_buffer, context);
+  }
+
+  if (!image) {
+    FML_DLOG(ERROR) << "Could not wrap Metal texture as a Skia image.";
+  }
+
+  return image;
+}
+
+sk_sp<SkImage> IOSExternalTextureMetal::WrapYUVAExternalPixelBuffer(
+    fml::CFRef<CVPixelBufferRef> pixel_buffer,
+    GrDirectContext* context) const {
   auto texture_size =
       SkISize::Make(CVPixelBufferGetWidth(pixel_buffer), CVPixelBufferGetHeight(pixel_buffer));
+  CVMetalTextureRef y_metal_texture_raw = NULL;
+  {
+    auto cv_return =
+        CVMetalTextureCacheCreateTextureFromImage(kCFAllocatorDefault,    // allocator
+                                                  texture_cache_,         // texture cache
+                                                  pixel_buffer,           // source image
+                                                  NULL,                   // texture attributes
```

Comment:
s/NULL/nullptr

Reply:
Since NULL was used in the original code, in order to keep the style consistent, I also used NULL, do I need to change all to nullptr?

Reply:
https://google.github.io/styleguide/cppguide.html#0_and_nullptr/NULL

Reply:
done